lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -914,8 +912,8 @@ JavaScriptCompiler.prototype = {
       let existing = this.matchExistingProgram(child);
 
       if (existing == null) {
-        this.context.programs.push(''); // Placeholder to prevent name conflicts for nested children
-        let index = this.context.programs.length;
+        // Placeholder to prevent name conflicts for nested children
+        let index = this.context.programs.push('') - 1;
```

Comment:
`push()` returns the length, smart optimization!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push#return_value

Thinking if we should keep the index at `1` and remove the `- 1`. :thinking: